doc/kernel-limitations.md

```diff
@@ -0,0 +1,174 @@
+# Notable Kernel limitations of Mystikos
```

Comment:
This doc primarily discusses  Mystikos-SGX security design. Most of  the covered security design topics are captured in Open Enclave design docs and the Mystikos implemented same design as Open Enclave.  If the doc is posited as a security design doc, it misses many other security design topics. I think for the syscall/kernel limitation docs, we can focus on the limitation, without the in-depth discussion of the security design choices behind the limitation. We can capture security design topics in separate docs.

Reply:
This doc tries to answer these questions: 
> As an application developer, what do I need to know regarding Mystikos kernel vs. Linux kernel in order to write a performant and safe program that can be lift-n-shifted into a TEE? And when it runs in a TEE, what could be the incompatibility from running on Linux?

These challenges require a holistic approach from the beginning. Security really should be a first consideration, and shouldn't be deferred to another project that doesn't have a kernel.

I think we can take out the security design choices but leave the security limitations in place.